install_config/persistent_storage/persistent_storage_glusterfs.adoc

```diff
@@ -203,22 +213,198 @@ gluster-default-volume   <none>    2147483648   RWX           Available
+and can only be referenced only by pods within that same project. Any attempt to
+access a PV across a project causes the pod to fail.
 
-[[gluster-volume-security]]
+The next step can be to create a persistent volume claim (PVC) which will bind
+to the new PV:
 
+.PVC Object Definition
+====
+[source,yaml]
+----
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: gluster-claim
+spec:
+  accessModes:
+  - ReadWriteMany <1>
+  resources:
+     requests:
+       storage: 1Gi <2>
+----
+<1> As mentioned above for PVs, the `*accessModes*` do not enforce security, but
+rather act as labels to match a PV to a PVC.
+<2> This claim will look for PVs offering *1Gi* or greater capacity.
+====
+
+Save the definition to a file, for example *_gluster-claim.yaml_*, and create the
+PVC:
+
+====
+----
+# oc create -f gluster-claim.yaml
+----
+====
+
+[NOTE]
+====
+PVs and PVCs are not necessary, just convenient, and make sharing a volume
+across a project simpler. The gluster-specific information contained in the PV
+definition can also be defined directly in a pod spec.
+====
+
+[[gluster-volume-security]]
 === Volume Security
-Users request storage with a `*PersistentVolumeClaim*`. This claim exists only
-in the user's namespace and can only be referenced by a pod within that same
-namespace. Any attempt to access a persistent volume across a namespace causes
-the pod to fail.
 
-Additionally, permissions and ownership can be controlled as normal by the
-Gluster server administrators using normal POSIX compliant security.
+This section covers Gluster volume security, including matching permissions and
+SELinux considerations. The reader is expected to understand the basics of POSIX
+permissions, process UIDs, supplemental groups, and SELinux.
+
+[NOTE]
+====
+See
+link:../../install_config/persistent_storage/pod_security_context.html[Volume
+Security] before implementing gluster volumes.
+====
+
+Assume that the target gluster volume, "_HadoopVol_", is mounted under
+_/mnt/glusterfs/_, with the following POSIX permissions and selinux labels:
+
+[[gfs-mount]]
+====
+----
+# ls -lZ /mnt/glusterfs/
+drwxrwx---. yarn hadoop system_u:object_r:fusefs_t:s0    HadoopVol
+
+# id yarn
+uid=592(yarn) gid=590(hadoop) groups=590(hadoop)
+----
+====
+
+In order to access the "_HadoopVol_" volume, containers must match the SELinux
+label, and either run with a UID of 592, or with 590 in their supplemental groups.
+The OpenShift GlusterFS plug-in mounts the volume in the container with the
+same POSIX ownership and permissions found on the target gluster mount, namely
+owner will be 592 and group id will be 590. However, the container is not run with
+its effective UID equal to the owner of the gluster mount, nor with its GID equal
+to the target volume's group id, which is the desired behavior. Instead, container
+UIDs and supplemental groups are controlled by Security Context Constraints
```

Comment:
Do you want to add some output from the scc here maybe? Or at least a link to that in the main document?